### PR TITLE
ENG-1080 Confluence to Slack demo

### PR DESCRIPTION
**Pull request by pashafateev**

### CODE REVIEW

File: confluence_to_slack/program.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
+"""A real-life workflow that integrates Confluence and Slack.
+
+Workflow:
+    1. Trigger: Detect a new page creation in Confluence space (specified
+    in the autokitteh.yaml).
+    2. Filter: Check if the page has the specified label.
+    3. Notify: Send a message to a Slack channel with a snippet of the page.
```
Comment (daabr) [inline confluence_to_slack/program.py:7]: Linguistic nit: the trigger is the creation, not the detection.

```suggestion
    1. Trigger: a new page is created in Confluence
    2. Static filter: the page is in a specific Confluence space
       (specified in the "autokitteh.yaml" manifest file)
    3. Runtime filter: check if the page has a specific label
    4. Notify: send a message to a Slack channel with a snippet of the page
```

File: confluence_to_slack/program.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+from autokitteh.atlassian import confluence_client
+from autokitteh.slack import slack_client
+
+
+AK_CONFLUENCE_CONNECTION = "my_confluence"
+AK_SLACK_CONNECTION = "my_slack"
+
+CONFLUENCE_LABEL = ""  # TODO: Replace with your Confluence label name
+SLACK_CHANNEL = ""  # TODO: Replace with your Slack channel name or ID
+SNIPPET_COUNT = 150
```
Comment (daabr) [inline confluence_to_slack/program.py:20]:
```suggestion
SNIPPET_LENGTH = 150
```

File: confluence_to_slack/program.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+
+def on_confluence_page_created(event):
+    """Workflow's entry-point."""
+
+    confluence = confluence_client(AK_CONFLUENCE_CONNECTION)
+    page_id = event.data.page.id
+
+    # filter out pages without the specified label
```
Comment (daabr) [inline confluence_to_slack/program.py:29]:
```suggestion
    # Ignore pages without the specified label.
```

File: confluence_to_slack/program.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    labels = confluence.get_page_labels(page_id)["results"]
+    if not any(label["name"] == CONFLUENCE_LABEL for label in labels):
+        return
+
+    # get the page body
```
Comment (daabr) [inline confluence_to_slack/program.py:34]:
```suggestion
    # Read the page body.
```

File: confluence_to_slack/program.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    res = confluence.get_page_by_id(page_id, expand="body.view, include-labels")
+    html_body = res["body"]["view"]["value"]
+
+    page = autokitteh.AttrDict(
+        {
+            "snippet": "```" + html_body[:SNIPPET_COUNT] + "\n```",
```
Comment (daabr) [inline confluence_to_slack/program.py:40]:
```suggestion
            "snippet": f"```{html_body[:SNIPPET_COUNT]}\n```",
```

**Review by daabr — APPROVED** [review]
Just nits, so approving. Nice!

File: confluence_to_slack/autokitteh.yaml [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+  connections:
+    - name: my_slack
+      integration: slack
+    - name: my_confluence
+      integration: confluence
+  triggers:
+    - name: confluence_page_created
+      connection: my_confluence
+      event_type: page_created
+      # TODO: Replace <your spaceKey> with the spaceKey of the Confluence space you want to monitor
```
[the remark this answers is not in the document]
Comment (pashafateev) [reply confluence_to_slack/autokitteh.yaml]: Done

**Pull request merged**